### PR TITLE
Set default link for images using image_default_link_type option 

### LINKED ISSUE #10173: image_default_link_type is ignored by Gutenberg
The setting image_default_link_type  from options.php is not used by Gutenberg: mine is set to "file", yet the editor defaults every image or gallery to the "none" link type. I have to set them all manually instead.

### PR DESCRIPTION
**Pull request by mkaz**
## Description

Use the option for image_default_link_type to set default link for images.
The related PR for galleries is #25582 

Fixes #10173

## How has this been tested?

1. Set the WordPress option `image_default_link_type` to "file" or "post"
You can set by going to your site's `wp-admin/options.php` page or via a plugin using a function like:

```
add_action( 'after_setup_theme', function() {
    update_option( 'image_default_link_type', 'file' );
});
```

2. Add an image in the editor and confirm the default link type is automatically selected.

confirm-media-link

## Types of changes

This unsets the default link type in Gutenberg none in favor of the WordPress option default. It checks to see if the option is set and sets the linkDestination based on the option.

There is an unfortunate mismtach between what Gutenberg thinks the option should be and what the historic option values are. The classic editor and previous use `file` to represent Media File, and `post` to represent "Attachment Page". The Gutenberg constants used are `media` and `attachment`.

### CODE REVIEW

File: packages/block-library/src/image/edit.js [hunk 1]
```diff
@@ -152,21 +155,46 @@ export function ImageEdit( {
...
-		if ( linkDestination === LINK_DESTINATION_MEDIA ) {
-			// Update the media link.
-			mediaAttributes.href = media.url;
+		// Check if default link setting should be used.
+		let linkDestination = attributes.linkDestination;
+		if ( ! linkDestination ) {
+			// Use the WordPress option to determine the proper default.
+			// The constants used in Gutenberg do not match WP options so a little more complicated than ideal.
+			// TODO: fix this in a follow up PR, requires updating media-text and ui component.
+			switch ( wp.media.view.settings.defaultProps.link ) {
```
Comment (ajlende) [inline packages/block-library/src/image/edit.js:164]: We probably want to guard against something in this chain being `undefined` if Gutenberg is running in a context that doesn't have it. Luckily we're using Babel, so we can use the optional chaining operator, `?.` and throw on a `default` case.

Comment (mkaz) [reply packages/block-library/src/image/edit.js]: Good point, it could fatal. I was thinking of null would work fine.
Addressed in: 89bfd84

**Comment by StevenDufresne** [comment]
Tested using "file" & "post" and this is working as expected.

**Review by ajlende — APPROVED** [review]
LGTM 👍

**Comment by KaziMahbuburRahman** [comment]
Thanks

**Comment by KaziMahbuburRahman** [comment]
can i disable images inserting from url?
I am using a plugin which helps me to upload image directly when I paste image from another site but Wordpress automatically add image from external source which I don't want...
can i turn off anyway?

**Pull request merged**